packages/diagrams/frontend/sirius-components-diagrams/src/renderer/palette/Palette.tsx

```diff
@@ -114,6 +119,14 @@ export const Palette = ({
     controlledPosition: { x: 0, y: 0 },
   });
 
+  const { hideDiagramPalette } = useDiagramPalette();
+  const { hideDiagramElementPalette } = useDiagramElementPalette();
+
+  const closeAllPalettes = useCallback(() => {
+    hideDiagramPalette();
+    hideDiagramElementPalette();
+  }, [hideDiagramPalette, hideDiagramElementPalette]);
```

Comment:
Same comment as usual, the palette should not depend on its parent, it should tell its parent component that it needs to be closed. We should rename its props `onEscape` to `onClose` and leverage that instead.